java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -296,98 +278,124 @@ private void readVector(Field field, FieldVector vector) throws JsonParseExcepti
-          byte[] value = decodeHexSafe(parser.readValueAs(String.class));
-          DecimalUtility.writeByteArrayToArrowBuf(value, decimalVector.getBuffer(), i);
-        }
-        break;
-      case VARBINARY:
-        ((VarBinaryVector) valueVector).getMutator().setSafe(i, decodeHexSafe(parser.readValueAs(String.class)));
-        break;
-      case VARCHAR:
-        ((VarCharVector) valueVector).getMutator().setSafe(i, parser.readValueAs(String.class).getBytes(UTF_8));
-        break;
-      case DATEDAY:
-        ((DateDayVector) valueVector).getMutator().set(i, parser.readValueAs(Integer.class));
-        break;
-      case DATEMILLI:
-        ((DateMilliVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESEC:
-        ((TimeSecVector) valueVector).getMutator().set(i, parser.readValueAs(Integer.class));
-        break;
-      case TIMEMILLI:
-        ((TimeMilliVector) valueVector).getMutator().set(i, parser.readValueAs(Integer.class));
-        break;
-      case TIMEMICRO:
-        ((TimeMicroVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMENANO:
-        ((TimeNanoVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPSEC:
-        ((TimeStampSecVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPMILLI:
-        ((TimeStampMilliVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPMICRO:
-        ((TimeStampMicroVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPNANO:
-        ((TimeStampNanoVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPSECTZ:
-        ((TimeStampSecTZVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPMILLITZ:
-        ((TimeStampMilliTZVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPMICROTZ:
-        ((TimeStampMicroTZVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      case TIMESTAMPNANOTZ:
-        ((TimeStampNanoTZVector) valueVector).getMutator().set(i, parser.readValueAs(Long.class));
-        break;
-      default:
-        throw new UnsupportedOperationException("minor type: " + valueVector.getMinorType());
+  private ArrowBuf setValueFromParser(ArrowVectorType bufferType, FieldVector vector,
+                                      ArrowBuf buffer, ArrowBuf offsetBuffer, int index, int valueCount) throws IOException {
+    if (bufferType.equals(TYPE)) {
+      buffer = NullableTinyIntVector.set(buffer, allocator,
+              valueCount, index, parser.readValueAs(Byte.class));
     }
+    else if (bufferType.equals(OFFSET)) {
```

Comment:
style: combine with line above

Reply:
Done.

Reply:
We probably want to run checkstyle on the new vector classes before merge.

Reply:
Maybe we ignore style until the change set are finalized and just fix it all at once.

Reply:
Yes, definitely. Once the patch is ready for merge to java-vector-refactor branch, we can check that. Or once we plan to push both patches from java-vector-refactor to master, then we can do and address the failures at one go.

Reply:
Yes, let's address that and any other cosmetic changes once we plan to push the two patches from java-vector-refactor branch to master